src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joined = [...this];
+    let joinedSring = '';
+
+    for (let i = 0; i < joined.length; i++) {
+      if (joined[i] === undefined || joined[i] === null) {
+        joined[i] = '';
+      }
+      joinedSring += String(joined[i]) + separator;
+    }
+
+    switch (separator) {
+      case '':
+        return joinedSring.slice(0, joinedSring.length);
+
+      default:
+        return joinedSring.slice(0, joinedSring.lastIndexOf(separator));
+    }
```

Comment:
Instead of using additional `switch / case` here, you can simply add some check inside the loop for checking should you add separator or no